sklearn/neural_network/tests/test_mlp.py

```diff
@@ -556,3 +562,45 @@ def test_adaptive_learning_rate():
     clf.fit(X, y)
     assert_greater(clf.max_iter, clf.n_iter_)
     assert_greater(1e-6, clf._optimizer.learning_rate)
+
+
+def test_warm_start():
+    X = X_iris[:150]
+    y = y_iris[:150]
+
+    y_2classes = np.array([0] * 75 + [1] * 75)
+    y_3classes = np.array([0] * 50 + [1] * 50 + [2] * 50)
+    y_4classes = np.array([0] * 37 + [1] * 37 + [2] * 38 + [3] * 38)
+    y_5classes = np.array([0] * 30 + [1] * 30 + [2] * 30 + [3] * 30 + [4] * 30)
+
+    with ignore_warnings(category=Warning):
+
+        # Test with 2 unique labels
+        clf = MLPClassifier(hidden_layer_sizes=2, solver='lbfgs',
+                            warm_start=True)
+        clf.fit(X, y)
+        message = ("`y` has classes not in `self.classes_`. `self.classes_`"
+                   " has [0 1 2]. 'y' has [0 1].")
+        assert_raise_message(ValueError, message, clf.fit, X, y_2classes)
+
+        # Test with 3 unique labels
+        clf = MLPClassifier(hidden_layer_sizes=2, solver='lbfgs',
+                            warm_start=True)
+        clf.fit(X, y)
+        clf.fit(X, y_3classes)
+
+        # Test with 4 unique labels
+        clf = MLPClassifier(hidden_layer_sizes=2, solver='lbfgs',
+                            warm_start=True)
+        clf.fit(X, y)
+        message = ("`y` has classes not in `self.classes_`. `self.classes_`"
+                   " has [0 1 2]. 'y' has [0 1 2 3].")
+        assert_raise_message(ValueError, message, clf.fit, X, y_4classes)
+
+        # Test with 5 unique labels
```

Comment:
This seems unnecessary. Instead can we test for `y` having labels [1, 2, 3] (no 0) or [0, 1, 3]: correct number of labels, incorrect set?